docs/design/kademlia-removal.md

```diff
@@ -0,0 +1,84 @@
+# Kademlia Removal
+
+## Abstract
+
+This design document outlines the communication protocol between satellites and 
+storage nodes, network refreshes, and kademlia removal given the satellite opt-in 
+capability for storage nodes. 
+
+## Background
+User research on how Storage Node Operators would connect to satellites resulted in a 
+simple solution called Opt-in SNO select. Satellites wait for storage nodes to work 
+with them, and storage nodes manually select the satellites they want to work with. 
+The initial implementation of this allows SNOs to update their trusted satellite list 
+in a hardcoded configuration file, but future improvements will enable users to manage 
+this list through a web console. Due to the nature of this change, the Storj network 
+at this point in time no longer requires the Kademlia DHT or related entities. 
+We will replace this protocol with direct communication between satellites and 
+storage nodes, and keep the network fresh without kademlia node discovery and random 
+lookups.
+
+## Design
+
+1. Nodes reach out to satellites that they want to work with
+    
+    - At the time of writing, a storage node operator can input a list of trusted satellite 
+    IDs and addresses into their configuration file on setup. There are several tardigrade-level 
+    satellites already listed as a preset. This list is imported into a Go map data structure 
+    within the “trust” package.
+    - Eventually during the design and implementation of trusted satellite management, 
+    this configured list may be imported into a on-disk data store, eg SQL table. Users will 
+    be able to modify this list through the storage node operator console. However, this is a 
+    separate discussion from this design document. This design is satisfied with a static trusted 
+    satellites list.
+    - Nodes will be able to communicate with satellites through the “transport” and “trust” packages 
+    directly rather than using kademlia to traverse the network to find the address of a given ID.
+
+2. Network refreshes at a regular interval
+    - Nodes will keep themselves up to date in the network by pinging all the satellites in their 
+    trusted list every hour. 
+    - Satellites will ping the nodes back
+        - If is it successful, the satellite will insert or update the node in the overlay cache and 
+        notify the node of success
+        - If not, it does not proceed with updating the overlay cache, and the node receives an error message
```

Comment:
A response from the server after updating the cache would mean a second communication from the server back to storage node (third overall).  Do you think there's value in rolling this feature into the initial ping of the satellite from storage node, where that ping would return a 'last status'?